src/core/include/openvino/op/search_sorted.hpp

```diff
@@ -22,7 +22,15 @@ class OPENVINO_API SearchSorted : public Op {
     /// \param values          Values to search indexs for.
     /// \param right_mode      If False, return the first suitable index that is found for given value. If True, return
     /// the last such index.
-    SearchSorted(const Output<Node>& sorted_sequence, const Output<Node>& values, bool right_mode = false);
+    /// \param output_type The element type of the output tensor. This is purely an implementation flag, which
+    /// is used to convert the output type for CPU plugin in ConvertPrecision transformation (and potentially other
+    /// plugins as well). Setting this flag to element::i32 will result in the output tensor of i32 element type.
+    /// Setting this flag to element::i64 will generally not give any effect, since it will be converted to i32 anyway,
+    /// at least for CPU plugin.
```

Comment:
Does it really not give any effect? Wouldn't it be converted back to i64 when the output is the result from the model?

Reply:
And what happens then if the output exceeds int32 capacity? Why would any one set it to i64, knowing what is going on under the hood?

Reply:
it's up to the CPU plugin or some another plugin which precision to use for this or that operation. 
ConvertPrecision transformation is used for this purpose in the plugins, the plugins configure which precision to use.
It also aligns mixed precision cases. 

Our business is to specify the operation. And this is a general approach of how we support operations that returns indices as an output. "output_type" should simplify the required precision conversion.

We shouldn't describe CPU plugin behavior in the operation.
This is the plugin business to use i64 or i32. 

So I wouldn't recommend to add this part
" This is purely an implementation flag, which
    /// is used to convert the output type for CPU plugin in ConvertPrecision transformation (and potentially other
    /// plugins as well). Setting this flag to element::i32 will result in the output tensor of i32 element type.
    /// Setting this flag to element::i64 will generally not give any effect, since it will be converted to i32 anyway,
    /// at least for CPU plugin.
"